libs/datamodel/core/src/transform/ast_to_dml/db/attributes.rs

```diff
@@ -560,29 +770,82 @@ fn model_unique<'ast>(
         is_unique: true,
         ..Default::default()
     };
-
     common_index_validations(args, &mut index_data, model_id, ctx);
 
+    let ast_model = &ctx.db.ast[model_id];
+    let model_db_name = data.mapped_name.unwrap_or(&ast_model.name.name);
+
+    let field_db_names: Vec<&str> = get_field_db_names(model_id, &index_data.fields, ctx);
+
+    let name = get_name_argument(args, ctx);
+
+    let (name, db_name) = if ctx.db.preview_features.contains(NamedConstraints) {
+        //todo compatibility hack
```

Comment:
Would this block release before it is done?

Reply:
The functionality is already as intended. I left two longer comments for the @@index and the @@unique case to explain what the compatibility behaviour is exactly.